TreeDB/cmd/treedb_rag_interleave/main_test.go

```
@@ -0,0 +1,135 @@
+	for _, arg := range []string{"-smoke", "--smoke", "--dir=/tmp/shared", "--product-base-sha=wrong"} {
+		t.Run(arg, func(t *testing.T) {
+			if _, err := parseConfig(append(append([]string(nil), base...), arg)); err == nil {
+				t.Fatalf("worker argument %q accepted", arg)
+			}
+		})
+	}
+}
+
+func TestDecodeWithTimeout(t *testing.T) {
+	reader, writer := io.Pipe()
+	defer reader.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `reader.Close` is not checked

(errcheck)